config/routes.rb

```diff
@@ -1,7 +1,18 @@
 Rails.application.routes.draw do
   get "/users", to: "users#index"
   get "/users/:id", to: "users#show"
-  get "/users/:id/posts", to: "posts#index"
-  get "/users/:id/posts/:id", to: "posts#show"
+  get "/users/:user_id/posts", to: "posts#index"
+  get "/users/:user_id/posts/:post_id", to: "posts#show"
+  get '/users/:id/posts/new', to: 'posts#new', as: 'new_post' # new
+  get '/users/:user_id/posts/:post_id', to: 'comments#new', as: 'new_comment' # new
+  get '/users/:user_id/posts/:post_id/new_like', to: 'likes#new', as: 'new_like' # new
+
+  post '/users/:id/posts/new', to: 'posts#create' # create
+  post '/users/:user_id/posts/:post_id', to: 'comments#create' # create
+  post '/users/:user_id/posts/:post_id/new_like', to: 'likes#create' # create
```

Comment:
- [ ] Most of the routes are not working check where the issue might be you are supposed to have a route for creating a new post `/users/:id/posts/new` check if you are not making a mistake in your routes
- [ ] A route for creating a new comment and for displaying comments for a specific post are not working `/users/:user_id/posts/:post_id/                                                   comments` `/users/:user_id/posts/:post_id(.:format)                                                          comments/new`

Reply:
- [ ] Make sure that like has a working route currently `/users/:user_id/posts/:post_id/new_like(.:format) ` doesn't work

Reply:
Addressed ✔